streams/src/test/java/org/apache/kafka/streams/processor/internals/GlobalStateTaskTest.java

```diff
@@ -252,6 +252,10 @@ public void shouldCheckpointOffsetsWhenStateIsFlushed() {
     @Test
     public void shouldNotCheckpointIfNotReceivedEnoughRecords() {
         globalStateTask.initialize();
+        // Reset after initialization since checkpointing should happen during initialization, KAFKA-18168
+        stateMgr.checkpointWritten = false;
+        stateMgr.flushed = false;
```

Comment:
Nit: it might be slightly better, to do this a little later in the test, ie, just before we call `globalStateTask.maybeCheckpoint()`? Test logic might be easier to reads for humans as (1) reset flags, (2) checkpoint, (3) verify flags.

```
        globalStateTask.update(record(topic1, 1, currentOffsetT1 + 9000L, "foo".getBytes(), "foo".getBytes()));
        time.sleep(flushInterval); // flush interval elapsed
        
        stateMgr.checkpointWritten = false;
        stateMgr.flushed = false;
        
        globalStateTask.maybeCheckpoint();
        
        assertEquals(offsets, stateMgr.changelogOffsets());
        assertFalse(stateMgr.flushed);
        assertFalse(stateMgr.checkpointWritten);
```

Similar in other tests.